src/test/clojure/clara/test_rules.clj

```diff
@@ -1183,75 +1183,85 @@
                                  [?t <- Temperature (= ?l location)]
                                  [:not [Cold (= temperature (:temperature ?t))]]]]])
 
-        s (mk-session [cold-not-match-temp] :cache false)
-        s-with-prior (mk-session [negation-with-prior-bindings] :cache false)
-        s-with-nested (mk-session [nested-negation-with-prior-bindings] :cache false)]
+        object-query (dsl/parse-query [] [[?o <- Object]])
 
-    (is (= [{}]
-           (-> s
-               (fire-rules)
-               (query cold-not-match-temp))))
+        s (mk-session [cold-not-match-temp object-query] :cache false)
+        s-with-prior (mk-session [negation-with-prior-bindings object-query] :cache false)
+        s-with-nested (mk-session [nested-negation-with-prior-bindings object-query] :cache false)
+
+        no-system-types? (fn [session]
```

Comment:
That makes sense to me that you want to test that our internal objects do not match user-provided rules/queries.  It'd probably be nice to have a comment above this that mentions that?